setup.py

```diff
@@ -22,30 +23,16 @@ def _read_me() -> str:
     return readme
 
 
-def _get_version() -> str:
-    # ops.__init__ needs to pull in ops.charm to work around a circular
-    # import so we can't import it here as that pulls in yaml which isn't
-    # necessarily there yet.
-    version = 'unknown'
-    with open("ops/__init__.py", "rt", encoding="utf8") as fh:
-        source = fh.read()
-    code = ast.parse(source)
-    for node in code.body:
-        if isinstance(node, ast.Assign):
-            targets = [i.id for i in node.targets]
-            if '__version__' in targets:
-                if isinstance(node.value, ast.Str):
-                    # Python < 3.8
-                    version = node.value.s
-                else:
-                    version = node.value.value
-                break
-    return version
-
+version = importlib.import_module('ops.version').version
+version_path = Path("ops/version.py")
+version_backup = Path("ops/version.py~")
+version_path.rename(version_backup)
+with version_path.open("wt", encoding="utf8") as fh:
+    print("# this is a generated file\n\nversion = {!r}".format(version), file=fh)
```

Comment:
Why not `fh.write("# this is a generated file\n\nversion = {!r}".format(version))`?

Reply:
see how you forgot the ending newline? that's why :-D

Reply:
I changed it to use a write, but also a multiline string. Everybody wins.